extkeys/hdkey.go

```diff
@@ -194,7 +194,13 @@ func (k *ExtendedKey) Child(i uint32) (*ExtendedKey, error) {
 		keyBigInt.Add(keyBigInt, parentKeyBigInt)
 		keyBigInt.Mod(keyBigInt, btcec.S256().N)
 
-		child.KeyData = keyBigInt.Bytes()
+		keyData := keyBigInt.Bytes()
+		if len(keyData) < 32 {
+			extra := make([]byte, 32-len(keyData))
```

Comment:
if you want you could do a single allocation instead of 2

```
extra := make([]byte, 32-len, 32)
```

Reply:
@dmitryn yes good point I'll do it!